golden-ratio.el

```diff
@@ -157,30 +151,32 @@ will prevent the window to be resized to the golden ratio."
       (when golden-ratio-recenter
         (scroll-right) (recenter)))))
 
-;; Should return nil
-(defadvice other-window
-    (after golden-ratio-resize-window)
-  (golden-ratio) nil)
-
-;; Should return the buffer
-(defadvice pop-to-buffer
-    (around golden-ratio-resize-window)
-  (prog1 ad-do-it (golden-ratio)))
-
-(defun golden-ratio--post-command-hook ()
-  (when (or (memq this-command golden-ratio-extra-commands)
-            (and (consp this-command) ; A lambda form.
-                 (loop for com in golden-ratio-extra-commands
-                       thereis (or (memq com this-command)
-                                   (memq (car-safe com) this-command)))))
-    ;; This is needed in emacs-25 to avoid this error from `recenter':
-    ;; `recenter'ing a window that does not display current-buffer.
-    ;; This doesn't happen in emacs-24.4 and previous versions.
-    (run-with-idle-timer 0.01 nil (lambda () (golden-ratio)))))
+(defvar golden-ratio--focus-alist '()
+  "Golden-ratio-selected window, per frame.")
+
+(defun golden-ratio--focus ()
+  (assq (selected-frame) golden-ratio--focus-alist))
+
+(defun golden-ratio--refocus ()
+  (lexical-let* ((frame (selected-frame))
```

Comment:
`lexical-let*` is obsolete. Just use `let*`, I don't think lexical bindings are required here.